File: Algorithms/minimum-depth-of-binary-tree.cpp
// https://leetcode.com/problems/minimum-depth-of-binary-tree/description/
class Solution {
public:
    int minDepth(TreeNode* root) {
        return root == NULL ? 0 : root->left == NULL ? 1 + minDepth(root->right) : root->right == NULL ? 1 + minDepth(root->left) : min(1 + minDepth(root->left), 1 + minDepth(root->right));
    }
};

File: Algorithms/minimum-depth-of-binary-tree.js
//https://leetcode.com/problems/minimum-depth-of-binary-tree/description/
var minDepth = function(root) {
    if(root === null) return 0;
    if(root.left === null) return 1 + minDepth(root.right);
    if(root.right === null) return 1 + minDepth(root.left);
    return Math.min(1 + minDepth(root.left), 1 + minDepth(root.right));
};
